Source/WebCore/platform/network/cocoa/NetworkStorageSessionCocoa.mm

```diff
@@ -172,15 +172,23 @@ - (CFURLRef)_cfurl;
     END_BLOCK_OBJC_EXCEPTIONS
 }
 
-RetainPtr<NSHTTPCookieStorage> NetworkStorageSession::nsCookieStorage() const
+RetainPtr<NSHTTPCookieStorage> NetworkStorageSession::nsCookieStorageIfExists() const
 {
     ASSERT(hasProcessPrivilege(ProcessPrivilege::CanAccessRawCookies) || m_isInMemoryCookieStore);
     auto cfCookieStorage = cookieStorage();
     ASSERT(cfCookieStorage || !m_isInMemoryCookieStore);
     if (!m_isInMemoryCookieStore && (!cfCookieStorage || [NSHTTPCookieStorage sharedHTTPCookieStorage]._cookieStorage == cfCookieStorage))
         return [NSHTTPCookieStorage sharedHTTPCookieStorage];
 
-    return adoptNS([[NSHTTPCookieStorage alloc] _initWithCFHTTPCookieStorage:cfCookieStorage.get()]);
```

Comment:
Why is the original code not checking if `cfCookieStorage` is `nullptr' before using it. I kept the same logic for now but am wondering if we should check that?

It doesn't seem like that's causing the crash though since the crash isn't a `nullptr` dereference.